tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Provide four composable, idempotent tools exposed via CLI and MCP agent: CI Update, Changelog Update, Version Update, Metadata Update.
+- [ ] Normalize execution order and expected outputs across all supported languages.
+- [ ] Encode management-plane vs data-plane behavioral differences explicitly (especially version & changelog logic).
+- [ ] Emit structured JSON results (status + message + next steps hints) enabling orchestration by AI agents and pipelines.
+- [ ] Support configuration-driven per-language scripting (path or inline) without recompiling the tool.
+
+### Exceptions and Limitations
+
+#### Exception 1: Data-Plane Deferred Version Bump
+
+**Description:** Data-plane packages typically defer version adjustment and release date update until the release preparedness phase, not immediately after Stage 6 baseline tasks.
+**Impact:** The `version update` tool may be a no-op for data-plane flows outside release prep.
+**Workaround:** Initial version of milestone 1 remains user decided and manual edits.
+
+#### Exception 2: Changelog Auto-Generation (Data-Plane)
+
+**Description:** No automatic generation for data-plane; manual editing expected.
+**Impact:** Tool returns success with a `next_steps` message prompting manual edits.
+**Workaround:** Future LLM integration (post milestone 1).
+
+#### Language-Specific Limitations
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| JavaScript | Code formatting isn't hooked up to the emitter | Noisy diffs or linter failures | Run code formatting step in the `update-metadata.ps1` script |
+
+---
+
+## Design Proposal
+
+### Overview
+
+Implement four individual CLI + MCP tools. Each tool:
+
+1. Resolves package path & language.
+2. Loads the configuration for the language.
+3. Chooses inline command over script path if both present.
+4. Executes the relevant sub-script(s) in an ordered, fault-isolated manner.
+5. Aggregates structured result JSON with a `result`, human-readable `message`, and optional `next_steps` hint guiding subsequent tool invocation.
+
+Provides a singular CLI + MCP command that orchestrates the four tools (update-ci → update-changelog → update-version → update-metadata) and returns a combined JSON summary.
+
+### Detailed Design
+
+The four tools are intentionally small, composable units. Each emits a machine-readable JSON payload to enable deterministic chaining by an agent or CI workflow. All tools share common behaviors:
+
+| Area | Expected Behavior |
+|--------|-------------------|
+| Invocation Mode | CLI and MCP (agent) alias; same semantics |
+| Input Path Validation | Must exist and contain a recognizable SDK language structure; errors early if not |
+| Config Resolution | Load `swagger_to_sdk_config.json`; prefer `inline command` over `script path` when both are configured |
+| Timeout (default) | 5 minutes per tool invocation |
```

Comment:
I REALLY hope that none of the 4 scenarios you listed takes more than seconds. What is the reason for 5min?